components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -48,22 +55,49 @@ public GradleEnterpriseApiClient(URL baseUrl, CustomValueNames customValueNames,
         this.logger = logger;
     }
 
-    private OkHttpClient configureProxyAuthentication(OkHttpClient httpClient) {
-        return httpClient.newBuilder()
-                .proxyAuthenticator((route, response) -> {
-                    if (response.code() == 407) {
-                        String scheme = response.request().url().scheme().toLowerCase(Locale.ROOT);
-                        String proxyUser = System.getProperty(scheme + ".proxyUser");
-                        String proxyPassword = System.getProperty(scheme + ".proxyPassword");
-                        if (proxyUser != null && proxyPassword != null) {
-                            return response.request().newBuilder()
-                                    .header("Proxy-Authorization", Credentials.basic(proxyUser, proxyPassword))
-                                    .build();
-                        }
-                    }
-                    return null;
-                })
-                .build();
+    private static final TrustManager TRUST_ALL_CERTS = new X509TrustManager() {
+        @Override
+        public void checkClientTrusted(java.security.cert.X509Certificate[] chain, String authType) {
+        }
+
+        @Override
+        public void checkServerTrusted(java.security.cert.X509Certificate[] chain, String authType) {
+        }
+
+        @Override
+        public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+            return new java.security.cert.X509Certificate[]{};
+        }
+    };
+
+    private OkHttpClient configureHttpClient(OkHttpClient httpClient) {
+        OkHttpClient.Builder builder = httpClient.newBuilder();
+
+        if ("1".equals(System.getenv(ALLOW_UNTRUSTED))) {
+            try {
+                SSLContext sslContext = SSLContext.getInstance("SSL");
+                sslContext.init(null, new TrustManager[]{TRUST_ALL_CERTS}, new java.security.SecureRandom());
+                builder.sslSocketFactory(sslContext.getSocketFactory(), (X509TrustManager) TRUST_ALL_CERTS);
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to configure SSL for ALLOW_UNTRUSTED", e);
+            }
+        }
```

Comment:
0: Minor, but I would probably extract configuring SSL and configuring proxy authentication into two separate methods.